src/wabt.cpp

```diff
@@ -520,21 +80,151 @@ ExecutionResult WabtEngine::execute(
 ) {
   HERA_DEBUG << "Executing with wabt...\n";
 
-  // This is the wasm state
+  // Set up the wabt Environment, which includes the Wasm store
+  // and the list of modules used for importing/exporting between modules
   interp::Environment env;
 
-  // Lets instantiate our state
+  // Set up interface to eei host functions
   ExecutionResult result;
-
   // FIXME: shouldn't have this loose pointer here, but needed for setWasmMemory
   WabtEthereumInterface* interface = new WabtEthereumInterface{context, state_code, msg, result, meterInterfaceGas};
 
-  // Lets add our host module
+  // Create host module
   // The lifecycle of this pointer is handled by `env`.
   interp::HostModule* hostModule = env.AppendHostModule("ethereum");
   heraAssert(hostModule, "Failed to create host module.");
-  hostModule->import_delegate = unique_ptr<WabtEthereumInterface>(interface);
 
+  hostModule->AppendFuncExport(
+    "useGas",
+    {{Type::I64}, {}},
```

Comment:
WABT handles the argument checking now. Here it is declared that `useGas()` takes single `i64` arg and returns nothing.